pkg/agent/flowexporter/exporter/exporter.go

```diff
@@ -0,0 +1,270 @@
+	nodeName := os.Getenv(nodeNameEnvKey)
+	if nodeName != "" {
+		return nodeName, nil
+	}
+	klog.Infof("Environment variable %s not found, using hostname instead", nodeNameEnvKey)
+	var err error
+	nodeName, err = os.Hostname()
+	if err != nil {
+		return "", fmt.Errorf("failed to get local hostname: %v", err)
+	}
+	return nodeName, nil
+}
+
+func genObservationID() (uint32, error) {
+	name, err := getNodeName()
+	if err != nil {
+		return 0, err
+	}
+	h := fnv.New32()
+	h.Write([]byte(name))
+	return h.Sum32(), nil
+}
+
+func InitFlowExporter(collector net.Addr, records flowrecords.FlowRecords) (*flowExporter, error) {
+	// Create IPFIX exporting expProcess and initialize registries and other related entities
+	obsID, err := genObservationID()
+	if err != nil {
+		return nil, fmt.Errorf("cannot generate obsID for IPFIX ipfixexport: %v", err)
+	}
+
+	expProcess, err := ipfix.NewIPFIXExportingProcess(collector, obsID)
+	if err != nil {
+		return nil, fmt.Errorf("error while initializing IPFIX exporting expProcess: %v", err)
+	}
+	expProcess.LoadRegistries()
+
+	flowExp := &flowExporter{
+		records,
+		expProcess,
+		nil,
+		0,
+	}
+
+	flowExp.templateID = flowExp.process.AddTemplate()
+	templateRec := ipfix.NewIPFIXTemplateRecord(uint16(len(IANAInfoElements)+len(AntreaInfoElements)), flowExp.templateID)
+
+	sentBytes, err := flowExp.sendTemplateRecord(templateRec)
+	if err != nil {
+		return nil, fmt.Errorf("error while creating and sending template record through IPFIX process: %v", err)
+	}
+	klog.V(2).Infof("Initialized flow exporter and sent %d bytes size of template record", sentBytes)
+
+	return flowExp, nil
+}
+
+func (exp *flowExporter) Run(stopCh <-chan struct{}) {
+	klog.Infof("Start exporting IPFIX flow records")
+	for {
+		select {
+		case <-stopCh:
+			exp.process.CloseConnToCollector()
+			break
+		case <-time.After(flowexporter.FlowExportInterval):
+			err := exp.flowRecords.BuildFlowRecords()
+			if err != nil {
+				klog.Errorf("Error when building flow records: %v", err)
+				return
+			}
+			err = exp.sendFlowRecords()
+			if err != nil {
+				klog.Errorf("Error when sending flow records: %v", err)
+				return
+			}
+		}
+	}
+}
+
+func (exp *flowExporter) sendFlowRecords() error {
+	err := exp.flowRecords.IterateFlowRecordsWithSendCB(exp.sendDataRecord, exp.templateID)
+	if err != nil {
+		return fmt.Errorf("error in iterating flow records: %v", err)
+	}
+	return nil
+}
+
+func (exp *flowExporter) sendTemplateRecord(templateRec ipfix.IPFIXRecord) (int, error) {
+	// Initialize this every time new template is added
+	exp.elementsList = make([]*ipfixentities.InfoElement, len(IANAInfoElements)+len(AntreaInfoElements))
+	// Add template header
+	_, err := templateRec.PrepareRecord()
+	if err != nil {
+		return 0, fmt.Errorf("error when writing template header: %v", err)
+	}
+
+	for i, ie := range IANAInfoElements {
+		var element *ipfixentities.InfoElement
+		var err error
+		if !strings.Contains(ie, "reverse") {
```

Comment:
I think you should add some comments to explain this or in IANAInfoElements?

Reply:
Also do you think more efficient to create another slice?

Reply:
Also it is more efficient to create another slice?

Reply:
Agreed. Will make reverse elements more explicit.